pkg/prometheus/operator.go

```diff
@@ -249,130 +249,112 @@ func New(conf Config, logger log.Logger, r prometheus.Registerer) (*Operator, er
-				return &cache.ListWatch{
-					ListFunc: func(options metav1.ListOptions) (object runtime.Object, err error) {
-						return mclient.MonitoringV1().Probes(namespace).List(context.TODO(), options)
-					},
-					WatchFunc: func(options metav1.ListOptions) (w watch.Interface, err error) {
-						return mclient.MonitoringV1().Probes(namespace).Watch(context.TODO(), options)
-					},
-				}
-			}),
+
+	c.probeInfs, err = informers.NewInformersForResource(
+		informers.NewMonitoringInformerFactories(
+			c.config.Namespaces.AllowList, c.config.Namespaces.DenyList,
+			mclient, resyncPeriod, nil,
 		),
-		&monitoringv1.Probe{}, resyncPeriod, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		monitoringv1.SchemeGroupVersion.WithResource(monitoringv1.ProbeName),
 	)
-	c.metrics.MustRegister(operator.NewStoreCollector("probe", c.probeInf.GetStore()))
-
-	c.ruleInf = cache.NewSharedIndexInformer(
-		c.metrics.NewInstrumentedListerWatcher(
-			listwatch.MultiNamespaceListerWatcher(c.logger, c.config.Namespaces.AllowList, c.config.Namespaces.DenyList, func(namespace string) cache.ListerWatcher {
-				return &cache.ListWatch{
-					ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-						return mclient.MonitoringV1().PrometheusRules(namespace).List(context.TODO(), options)
-					},
-					WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-						return mclient.MonitoringV1().PrometheusRules(namespace).Watch(context.TODO(), options)
-					},
-				}
-			}),
+
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating podmonitor informers")
+	}
+
+	c.ruleInfs, err = informers.NewInformersForResource(
+		informers.NewMonitoringInformerFactories(
+			c.config.Namespaces.AllowList, c.config.Namespaces.DenyList,
+			mclient, resyncPeriod, nil,
 		),
-		&monitoringv1.PrometheusRule{}, resyncPeriod, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		monitoringv1.SchemeGroupVersion.WithResource(monitoringv1.PrometheusRuleName),
 	)
-	c.metrics.MustRegister(operator.NewStoreCollector("prometheurule", c.ruleInf.GetStore()))
-
-	c.cmapInf = cache.NewSharedIndexInformer(
-		c.metrics.NewInstrumentedListerWatcher(
-			listwatch.MultiNamespaceListerWatcher(c.logger, c.config.Namespaces.PrometheusAllowList, c.config.Namespaces.DenyList, func(namespace string) cache.ListerWatcher {
-				return &cache.ListWatch{
-					ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-						options.LabelSelector = labelPrometheusName
-						return c.kclient.CoreV1().ConfigMaps(namespace).List(context.TODO(), options)
-					},
-					WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-						options.LabelSelector = labelPrometheusName
-						return c.kclient.CoreV1().ConfigMaps(namespace).Watch(context.TODO(), options)
-					},
-				}
-			}),
+
+	if err != nil {
```

Comment:
Let's be consistent with the cuddling of error checks

Reply:
fixed